app/services/ui_notifications/hosts/destroy.rb

```diff
@@ -10,12 +10,21 @@ def create
         Notification.create!(
           initiator: initiator,
           audience: audience,
-          # note we do not store the subject, as the object is being deleted.
+          subject: subject.owner, # note we store the host owner, as the host object is being deleted.
```

Comment:
this means we need to depend: delete on the user/usergroup? (not to delete a notification that belongs to a group that is being deleted)?

Reply:
@ohadlevy @timogoebel  I think it's the opposite, delete the notifications associated with that User/User group when they are deleted. E.g:

1. Host with owner "Ohad" is deleted.
2. Admin tries to remove user "Ohad" - gets a warning "User has notifications, cannot delete this user"
3. Admin logs in as "Ohad", marks all notifications as read
4. Admin logs back in as "admin" then removes "Ohad".

Whereas we can get by with:

1. Host with owner "Ohad" is deleted.
2. Admin removes user "Ohad" - the notifications are deleted too, they are irrelevant now. Neither the host nor the owner are in Foreman anymore. There's an Audit page if the admin wonders what happened with the host created in step 1.

Reply:
Sorry, I don't follow what this is about.
When a user that has notifications is deleted, the notifications are deleted as well (`dependant :destroy`).
But to be honest: Isn't this totally unrelated to this PR and worth a separate issue if there is still a mistake?

Reply:
@timogoebel I agree and I think that's the correct behavior - that's the 2nd part of my message ("Whereas we can get by with...")